pkg/services/preference/prefimpl/pref.go

```diff
@@ -141,6 +146,13 @@ func (s *Service) Patch(ctx context.Context, cmd *pref.PatchPreferenceCommand) e
 		exists = true
 	}
 
+	if cmd.Locale != nil {
+		if preference.JSONData == nil {
+			preference.JSONData = &pref.PreferenceJSONData{}
+		}
+		preference.JSONData.Locale = *cmd.Locale
```

Comment:
is there a case when cmd.Locale is nil?

Reply:
Not sure if that's new or was there when you commented, but the if right above this makes that check, so not on this line.